From e719822ed6cac32c213c81cc0bfb342266ff761f Mon Sep 17 00:00:00 2001
From: Liu Xiaowen <b37945@freescale.com>
Date: Wed, 14 Sep 2011 15:45:38 +0800
Subject: [PATCH 179/635] ENGR00156707 improve camera stability

when switching between camcorder and capture mode, the dequeue buffer
may run fast than queue. Then, it may cause dequeue failed.

Signed-off-by: Liu Xiaowen <b37945@freescale.com>
---
 mx5x/libcamera/CameraHal.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mx5x/libcamera/CameraHal.cpp b/mx5x/libcamera/CameraHal.cpp
index 6ab9974..9a350c5 100755
--- a/mx5x/libcamera/CameraHal.cpp
+++ b/mx5x/libcamera/CameraHal.cpp
@@ -688,6 +688,7 @@ namespace android {
         status_t ret = NO_ERROR;
         Mutex::Autolock lock(mLock);
         //isPreviewFinsh = 0;
+        mEnqueuedBufs = 0;
         if (mPreviewRunning) {
             return NO_ERROR;
         }
@@ -1805,6 +1806,7 @@ Pic_out:
                 }
                 mCaptureBuffers[queue_back_index].buf_state = WINDOW_BUFS_DEQUEUED;
                 nCameraBuffersQueued++;
+                mEnqueuedBufs --;
                 sem_post(&avab_dequeue_frame);
             }else {
                 CAMERA_HAL_ERR("dequeue invalide buffer!!!!");
-- 
1.8.0

